File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        head = 0
        tail = len(numbers) - 1
        while (head < tail):
            s = numbers[head] + numbers[tail]
            diff = target - s
            if diff == 0:
                return [head + 1, tail + 1]
            if diff > 0:
                head += 1
            else:
                tail -= 1

        return []

File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> m = new HashMap<>();
        int first = 0;
        int second = 0;
        for (int i = 0; i < nums.length; i++) {
            if (m.get(target - nums[i]) != null) {
                if (i < m.get(target - nums[i])) {
                    first = i;
                    second = m.get(target - nums[i]);
                } else {
                    second = i;
                    first = m.get(target - nums[i]);
                }
                
            }
            else {
                m.put(nums[i], i);
            }
        }
        int [] result = {first, second};
        return result;
    }
}
